### PR TITLE
feat: general implement

### PR DESCRIPTION
**Pull request by Zxilly**
ref to:
https://github.com/casbin/casbin-oa/blob/master/web/src/auth/Auth.js

@hsluoyz @Kininaru plz review

Signed-off-by: Zxilly <zhouxinyu1001@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Kininaru @sh1luo plz review

**Review by sh1luo — CHANGES_REQUESTED** [review]
I think this pr should be modified into one commit instead of two

File: src/sdk.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+class Sdk {
+    private config: sdkConfig
+
+    constructor(config: sdkConfig) {
+        this.config = config
+    }
+
+    public getAuthorizeUrl():string {
+        const redirectUri = `${window.location.origin}/callback`;
```
Comment (Kininaru) [inline src/sdk.ts:33]: I think the redirect url should be a parameter that users can customize.

**Review by Kininaru — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Zxilly we have updated some API names, see: https://github.com/casbin/casbin-oa/commit/d67a3f5263d9c78441675e59da8c5e0a00b3af71

Plz update this PR.

**Comment by Zxilly** [comment]
@hsluoyz updated

**Pull request merged**